src/cookbooks/client_guidelines.md

```diff
@@ -0,0 +1,137 @@
+# Client Implementation Guidelines for Experiments
+
+There are three supported approaches to developing experimental features for Firefox:
+
+- [Feature Gates](#feature-gates)
+  - The preferred method for testing experimental features is to __land in-tree__ and ship behind a __feature gate__.
+- [Firefox Prefs](#prefs)
+  - If feature gates are [not appropriate](#feature-gate-caveats) for your project, Firefox Preferences (AKA "__Prefs__") should be used.
+- [Firefox Extensions](#extensions) AKA "__Add-ons__".
+  - If the feature being tested should not land in the tree, or if it will ultimately ship as an extension, then an extension should be used.
+
+New features go through the standard Firefox review, testing, and deployment processes, and are then enabled experimentally in the field using [Normandy](https://github.com/mozilla/normandy).
+
+## Feature Gates
+
+A new Feature Gate library for Firefox Desktop is now available.
+
+__Each feature gate should represent a different experimental treatment__.
+
+### Feature Gate caveats
+
+The current Feature Gate library comes with a few caveats, and may not be appropriate for your situation:
+
+- Only JS is supported.
+- Always asynchronous.
+
+Future versions of the Feature Gate API will include C++/Rust support and a synchronous API.
+
+### Using the Feature Gate library
+
+Read [the documentation][feature-gate-docs] to get started.
+
+## Prefs
+
+Firefox Preferences (AKA "prefs") are commonly used to enable and disable features. However, prefs are more complex to implement correctly than [feature gates](#feature-gates).
+
+__Each pref should represent a different experimental treatment__.
```

Comment:
Probably will end up same as above. There is a really strong desire to allow multi-pref flips, to the point where folks are looking at https://github.com/nhnt11/multipreffer for upcoming experiments rather than using Normandy directly... not sure if we should try to work this in now or not, it hasn't actually been used yet.

Reply:
So, we might have to compromise on this point.

Reply:
The more I think about this, we should soften the language here and mention multipreffer, and that Normandy will have support for this "soon".